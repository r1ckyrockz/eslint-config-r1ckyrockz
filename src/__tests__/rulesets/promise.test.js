const { createPromiseRules } = require('../../rulesets/promise');

describe('createPromiseRules', () => {
  test('matches snapshot without typescript', () => {
    const project = {
      typescript: {
        hasTypeScript: false,
      },
    };

    expect(createPromiseRules(project)).toMatchSnapshot();
  });

  test('matches snapshot with typescript', () => {
    const project = {
      typescript: {
        hasTypeScript: true,
      },
    };

    expect(createPromiseRules(project)).toMatchSnapshot();
  });

  test('allows passing custom rules', () => {
    const rule = 'promise/always-return';
    const level = 'error';

    const project = {
      rules: {
        [rule]: level,
      },
      typescript: {
        hasTypeScript: true,
      },
    };

    const defaultRuleset = createPromiseRules({
      ...project,
      rules: {},
    });

    const result = createPromiseRules(project);

    expect(result[rule]).toBe(level);
    expect(result[rule]).not.toBe(defaultRuleset[rule]);

    expect(result).toMatchSnapshot();
  });
});
