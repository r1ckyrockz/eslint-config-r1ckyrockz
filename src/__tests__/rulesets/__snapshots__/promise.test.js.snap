// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createPromiseRules allows passing custom rules 1`] = `
Object {
  "promise/always-return": "error",
  "promise/avoid-new": "off",
  "promise/catch-or-return": "off",
  "promise/no-callback-in-promise": "off",
  "promise/no-native": "off",
  "promise/no-nesting": "warn",
  "promise/no-new-statics": "error",
  "promise/no-promise-in-callback": "warn",
  "promise/no-return-in-finally": "warn",
  "promise/no-return-wrap": "error",
  "promise/param-names": "warn",
  "promise/prefer-await-to-callbacks": "warn",
  "promise/prefer-await-to-then": "warn",
  "promise/valid-params": "off",
}
`;

exports[`createPromiseRules matches snapshot with typescript 1`] = `
Object {
  "promise/always-return": "off",
  "promise/avoid-new": "off",
  "promise/catch-or-return": "off",
  "promise/no-callback-in-promise": "off",
  "promise/no-native": "off",
  "promise/no-nesting": "warn",
  "promise/no-new-statics": "error",
  "promise/no-promise-in-callback": "warn",
  "promise/no-return-in-finally": "warn",
  "promise/no-return-wrap": "error",
  "promise/param-names": "warn",
  "promise/prefer-await-to-callbacks": "warn",
  "promise/prefer-await-to-then": "warn",
  "promise/valid-params": "off",
}
`;

exports[`createPromiseRules matches snapshot without typescript 1`] = `
Object {
  "promise/always-return": "off",
  "promise/avoid-new": "off",
  "promise/catch-or-return": Array [
    "error",
    Object {
      "allowFinally": true,
      "allowThen": true,
    },
  ],
  "promise/no-callback-in-promise": "off",
  "promise/no-native": "off",
  "promise/no-nesting": "warn",
  "promise/no-new-statics": "error",
  "promise/no-promise-in-callback": "warn",
  "promise/no-return-in-finally": "warn",
  "promise/no-return-wrap": "error",
  "promise/param-names": "warn",
  "promise/prefer-await-to-callbacks": "warn",
  "promise/prefer-await-to-then": "warn",
  "promise/valid-params": "error",
}
`;
